apps/kg/README.md

```diff
@@ -56,6 +56,21 @@ DGL-KE provides five knowledge graphs:
 
 Users can specify one of the datasets with `--dataset` in `train.py` and `eval.py`.
 
+DGL-KE also support user defined dataset. It supports two formats:
+
+- raw_udd, raw user defined dataset format. it should contains five files:
+  - *entities* stores the mapping between entity name and entity Id
+  - *relations* stores the mapping between relation name relation Id
+  - *train* stores the triples in the training set. In format [src_name, rel_name, dst_name]
+  - *valid* stores the triples in the validation set. In format [src_name, rel_name, dst_name]
+  - *test* stores the triples in the test set. In format [src_name, rel_name, dst_name]
+- udd, user defined dataset format. it should contains five files:
+  - *entities* stores the mapping between entity name and entity Id
+  - *relations* stores the mapping between relation name relation Id
+  - *train* stores the triples in the training set. In format [src_id, rel_id, dst_id]
+  - *valid* stores the triples in the validation set. In format [src_id, rel_id, dst_id]
+  - *test* stores the triples in the test set. In format [src_id, rel_id, dst_id]
```

Comment:
the difference is that one provides triplets of entity names and relation names. the other provides triplets of ids?

Reply:
Yes, now we have udd_raw with only triplets as input and udd with entity map/relation map and triplets as input